pennylane/templates/subroutines/double_excitation_unitary.py

```diff
@@ -0,0 +1,550 @@
+    operator entering the UCCSD ansatz.
+
+    .. math::
+
+        \hat{U}_{pqrs}^{(8)}(\theta) = \mathrm{exp} \Big\{ -\frac{i\theta}{8}
+        \bigotimes_{b=s+1}^{r-1} \hat{Z}_b \bigotimes_{a=q+1}^{p-1} \hat{Z}_a
+        (\hat{Y}_s \hat{Y}_r \hat{X}_q \hat{Y}_p) \Big\}
+
+    Args:
+        weight (float): angle :math:`\theta` entering the Z rotation acting on wire ``p``
+        s (int): qubit index ``s``
+        r (int): qubit index ``r``
+        q (int): qubit index ``q``
+        p (int): qubit index ``p``
+        set_cnot_wires (sequence[int]): two-element sequence with the indices of the qubits
+            the CNOT gates act on
+    """
+
+    # U_1, U_2, U_3, U_4 acting on wires 's', 'r', 'q' and 'p'
+    RX(-np.pi / 2, wires=s)
+    RX(-np.pi / 2, wires=r)
+    Hadamard(wires=q)
+    RX(-np.pi / 2, wires=p)
+
+    # Applying CNOTs between wires 's' and 'p'
+    for cnot_wires in set_cnot_wires:
+        CNOT(wires=cnot_wires)
+
+    # Z rotation acting on wire 'p'
+    RZ(-weight / 8, wires=p)
+
+    # Applying CNOTs in reverse order
+    for cnot_wires in reversed(set_cnot_wires):
+        CNOT(wires=cnot_wires)
+
+    # U_1^+, U_2^+, U_3^+, U_4^+ acting on wires 's', 'r', 'q' and 'p'
+    RX(np.pi / 2, wires=s)
+    RX(np.pi / 2, wires=r)
+    Hadamard(wires=q)
+    RX(np.pi / 2, wires=p)
+
+
+@template
+def DoubleExcitationUnitary(weight, wires=None):
+    r"""Circuit to exponentiate the tensor product of Pauli matrices representing the
+    fermionic double-excitation operator entering the Unitary Coupled-Cluster Singles
+    and Doubles (UCCSD) ansatz. UCCSD is a VQE ansatz commonly used to run quantum
+    chemistry simulations.
+
+    The CC double-excitation operator is given by
+
+    .. math::
+
+        \hat{U}_{pqrs}(\theta) = \mathrm{exp} \{ \theta (\hat{c}_p^\dagger \hat{c}_q^\dagger
+        \hat{c}_r \hat{c}_s - \mathrm{H.c.}) \},
+
+    where :math:`\hat{c}` and :math:`\hat{c}^\dagger` are the fermionic annihilation and
+    creation operators and the indices :math:`r, s` and :math:`p, q` run over the occupied and
+    unoccupied molecular orbitals, respectively. Using the `Jordan-Wigner transformation
+    <https://arxiv.org/abs/1208.5986>`_ the fermionic operator defined above can be written
+    in terms of Pauli matrices (for more details see
+    `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_):
+
+    .. math::
+
+        \hat{U}_{pqrs}(\theta) = \mathrm{exp} \Big\{
+        \frac{i\theta}{8} \bigotimes_{b=s+1}^{r-1} \hat{Z}_b \bigotimes_{a=q+1}^{p-1}
+        \hat{Z}_a (\hat{X}_s \hat{X}_r \hat{Y}_q \hat{X}_p +
+        \hat{Y}_s \hat{X}_r \hat{Y}_q \hat{Y}_p + \hat{X}_s \hat{Y}_r \hat{Y}_q \hat{Y}_p +
+        \hat{X}_s \hat{X}_r \hat{X}_q \hat{Y}_p - \mathrm{H.c.}  ) \Big\}
+
+    The quantum circuit to exponentiate the tensor product of Pauli matrices entering
+    the latter equation is shown below:
+
+    .. figure:: ../../_static/templates/subroutines/double_excitation_unitary.png
```

Comment:
```suggestion
    the latter equation is shown below:
    |

    .. figure:: ../../_static/templates/subroutines/double_excitation_unitary.png

    |
```

Adding some space around the figure?